src/api/app/helpers/model_helper.rb

```diff
@@ -11,12 +11,14 @@ module ModelHelper
   def sync_hash_with_model(entry_class, dblist, inhasharray)
 
     keys = entry_class._sync_keys
-    oldentries = {}
+    entries = {}
+    to_delete = {}
 
     dblist.each do |e|
       key = ""
       keys.each{|k| key << "#{e.send(k).to_s}::"}
-      oldentries[key]=e
+      entries[key]=e
+      to_delete[key]=e
     end
 
```

Comment:
Just a proposal... ```to_delete = entries.clone``` might make it more obvious for readers that this is a copy of the entries hash